### PR TITLE
[Feature] Update Iceberg to 1.0 and add REST catalog support

### PR DESCRIPTION
**Pull request by bryanck**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Many Iceberg-based systems are moving from the Hive metastore to the recently added REST catalog. This PR updates Iceberg dependencies to version 1.0.0, and adds support for Iceberg's REST catalog. This will allow StarRocks to work with systems that implement the Iceberg REST catalog API.

## Checklist:

- [X] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 58 (12.07%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/hive/CachedClientPool.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/CachedGlueClientPool.java|0|1|00.00%|[56]|
|:large_blue_circle:|com/starrocks/connector/iceberg/CatalogLoader.java|0|13|00.00%|[45, 150, 151, 152, 153, 154, 155, 159, 160, 165, 166, 167, 168]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|8|00.00%|[70, 107, 280, 281, 282, 283, 284, 285]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/GlueClientPool.java|0|1|00.00%|[27]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|1|7|14.29%|[58, 59, 60, 101, 115, 116]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergRESTCatalog.java|3|20|15.00%|[29, 30, 31, 32, 33, 35, 45, 51, 56, 57, 62, 63, 65, 70, 71, 72, 73]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|5|20.00%|[95, 96, 200, 201]|
|:large_blue_circle:|com/starrocks/connector/iceberg/hive/HiveClientPool.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergCatalogType.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergRESTCatalog.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import static com.starrocks.connector.iceberg.IcebergUtil.convertToSRDatabase;
+
+public class IcebergRESTCatalog extends RESTCatalog implements IcebergCatalog {
+
+    private static final ConcurrentHashMap<String, IcebergRESTCatalog> REST_URI_TO_CATALOG =
+            new ConcurrentHashMap<>();
+
+    public static synchronized IcebergRESTCatalog getInstance(Map<String, String> properties) {
+        String uri = properties.get(CatalogProperties.URI);
+        if (!REST_URI_TO_CATALOG.containsKey(uri)) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergRESTCatalog.java:30]: use `computeIfAbsent` instead.

**Review by Youngwb — DISMISSED** [review]

**Comment by stephen-shelby** [comment]
Thanks for your contribution. I want to know how you test the rest catalog based on StarRocks.

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergRESTCatalog.java (hunk 1 above)
Comment (bryanck) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergRESTCatalog.java:30]: Yeah I agree, I was keeping it consistent w/ the other catalogs but I'll update it here

Comment (bryanck) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergRESTCatalog.java]: This is done.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 58 (17.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/hive/CachedClientPool.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/CachedGlueClientPool.java|0|1|00.00%|[56]|
|:large_blue_circle:|com/starrocks/connector/iceberg/CatalogLoader.java|0|13|00.00%|[45, 150, 151, 152, 153, 154, 155, 159, 160, 165, 166, 167, 168]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|8|00.00%|[69, 106, 278, 279, 280, 281, 282, 283]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/GlueClientPool.java|0|1|00.00%|[27]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|1|7|14.29%|[58, 59, 60, 101, 115, 116]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergRESTCatalog.java|3|19|15.79%|[29, 30, 33, 34, 35, 46, 52, 57, 58, 63, 64, 66, 71, 72, 73, 74]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|4|6|66.67%|[95, 96]|
|:large_blue_circle:|com/starrocks/connector/iceberg/hive/HiveClientPool.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergCatalogType.java|1|1|100.00%|[]|

**Comment by bryanck** [comment]
> Thanks for your contribution. I want to know how you test the rest catalog based on StarRocks.

We have a Docker image that will start up a REST catalog listening on port 8888 backed by a JDBC catalog if you want to give it a try: https://hub.docker.com/r/tabulario/iceberg-rest-jdbc/tags

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
> > Thanks for your contribution. I want to know how you test the rest catalog based on StarRocks.
> 
> We have a Docker image that will start up a REST catalog listening on port 8181 backed by a JDBC catalog if you want to give it a try: https://hub.docker.com/repository/docker/tabulario/iceberg-rest. Run `docker run -p 8181:8181 tabulario/iceberg-rest:latest`, then set the `uri` catalog property to `http://0.0.0.0:8181`

The latest tag is arm64 that our dev machine don't support. I use the 0.1.0 amd64 to test and found that when I execute `show databases from iceberg_catalog` in sr , it will return empty. Is this as expected?. I have executed follow command string:
`create external catalog iceberg_catalog properties ("type"="iceberg","iceberg.catalog.type"="rest","uri"="http://0.0.0.0:8181");`
`show databases from iceberg_catalog`

**Comment by bryanck** [comment]
Thanks for pointing out the issue with the image tag, I fixed the `latest` tag so it is multi-architecture now. The default config will be in-memory, so it will be empty on startup and will be cleared on restart. You can also use another catalog as a backend, there is an example of using Glue in the docs.

**Review by stephen-shelby — APPROVED** [review]

**Comment by stephen-shelby** [comment]
> 

I have tested the rest catalog by spark writing and StarRocks reading. Thanks~

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**